entities.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from abc import ABC, abstractmethod
+
+
+class Position(object):
+
+    def __init__(self, x, y):
+        self.x = x
+        self.y = y
+
+    def move(self, x, y):
+        self.x += x
+        self.y += y
+
+    def __repr__(self):
+        return f"({self.x}, {self.y})"
+
+
+STARTING_PLAYER_POS = Position(0, 0)
+
+
+class Entity(ABC):
+
+    @abstractmethod
+    def __init__(self, name, pos, type):
+        self.name = name
+        self.pos = pos
+        self.type = type
+
+    def __repr__(self):
+        return f"{self.name} <{self.type}>@{self.pos}"
+
+
+class Character(Entity, ABC):
+
+    @abstractmethod
+    def __init__(self, name, pos, type):
+        super().__init__(name, pos, type)
+        # base stats
+        self._strength = 5
+        self._constitution = 5
+        self._intelligence = 5
+        self._dexterity = 5
+        self._luck = 5
+        # computed stats
+        self._max_hp = self._compute_max_hp()
+        self._max_mp = self._compute_max_mp()
+        self._physical_dmg = self._compute_physical_damage()
+        self._ranged_dmg = self._compute_ranged_damage()
+        self._magical_dmg = self._compute_magical_damage()
+        self._hit_rate = self._compute_hit_rate()
+        self._dodge_rate = self._compute_dodge_rate()
```

Comment:
We might need to change all of these when we discuss the game mechanics, but it's OK to have them for now as a basic template.